src/main/java/com/aws/greengrass/shadowmanager/util/JsonMerger.java

```diff
@@ -44,6 +44,9 @@ public static void merge(JsonNode source, final JsonNode patch) throws InvalidRe
             return;
         }
 
+        // If the source node contains keys not present in the patch node, those keys have been deleted
+        removeDeletedFields((ObjectNode) source, (ObjectNode) patch);
```

Comment:
I'm concerned that this method (merge) may be used by multiple code paths with different types of shadow documents. Whether the "patch" is truly a patch, or if it is supposed to be the complete shadow document.

Have you checked to see the code paths that this is used to ensure that we're doing the right thing?

If this method is used to merge in a delta change, then that should not result in fields being removed since the delta is only including changed fields rather than all fields.

Reply:
That's something that just occurred to me also, going to dig into that a bit more to make sure this isn't breaking any other functionality

Reply:
Ah looks like this path is used for other merges, I'll look into another approach for this